src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestDiagnosticListener.net461.cs

```diff
@@ -1,163 +0,0 @@
-                return;
-            }
-
-            this.Tracer.StartActiveSpanFromActivity(Constants.HttpSpanPrefix + request.Method, activity, SpanKind.Client, out var span);
-
-            if (span.IsRecording)
-            {
-                span.PutComponentAttribute("http");
-                span.PutHttpMethodAttribute(request.Method);
-                span.PutHttpHostAttribute(request.RequestUri.Host, request.RequestUri.Port);
-                span.PutHttpRawUrlAttribute(request.RequestUri.OriginalString);
-
-                if (this.options.SetHttpFlavor)
-                {
-                    span.PutHttpFlavorAttribute(request.ProtocolVersion.ToString());
-                }
-            }
-
-            if (!(this.options.TextFormat is TraceContextFormat))
-            {
-                this.options.TextFormat.Inject<HttpWebRequest>(span.Context, request, (r, k, v) => r.Headers.Add(k, v));
-            }
-        }
-
-        public override void OnStopActivity(Activity activity, object payload)
-        {
-            const string EventNameSuffix = ".OnStopActivity";
-            var span = this.Tracer.CurrentSpan;
-            try
-            {
-                if (span == null || !span.Context.IsValid)
-                {
-                    InstrumentationEventSource.Log.NullOrBlankSpan(nameof(HttpWebRequestDiagnosticListener) + EventNameSuffix);
-                    return;
-                }
-
-                if (span.IsRecording)
-                {
-                    if (this.stopResponseFetcher.Fetch(payload) is HttpWebResponse response)
-                    {
-                        span.PutHttpStatusCode((int)response.StatusCode, response.StatusDescription);
-                    }
-                }
-            }
-            finally
-            {
-                span?.End();
-            }
-        }
-
-        public override void OnException(Activity activity, object payload)
-        {
-            const string EventNameSuffix = ".OnException";
-            var span = this.Tracer.CurrentSpan;
-            try
-            {
-                if (span == null || !span.Context.IsValid)
-                {
-                    InstrumentationEventSource.Log.NullOrBlankSpan(nameof(HttpWebRequestDiagnosticListener) + EventNameSuffix);
-                    return;
-                }
-
-                if (span.IsRecording)
-                {
-                    if (!(this.stopExceptionFetcher.Fetch(payload) is Exception exc))
-                    {
-                        InstrumentationEventSource.Log.NullPayload(nameof(HttpWebRequestDiagnosticListener) + EventNameSuffix);
-                        return;
-                    }
-
-                    ProcessException(span, exc);
-                }
-            }
-            finally
-            {
-                span?.End();
-            }
-        }
-
-        private static void ProcessException(TelemetrySpan span, Exception exception)
-        {
```

Comment:
Previous exception handling is here. Three cases: 1) We have a response code (brought forward), 2) We have a WebExceptionStatus (gap), and 3) Unknown exception (kind of brought forward, I set the exception.Message as a tag).